hummingbot/wallet/ethereum/watcher/account_balance_watcher.py

```diff
@@ -101,19 +104,21 @@ def address(self) -> str:
     def get_raw_balances(self) -> Dict[str, int]:
         return self._raw_account_balances.copy()
 
-    def get_all_balances(self) -> Dict[str, float]:
-        return dict((asset_name, self._raw_account_balances[asset_name] * math.pow(10, -self.get_decimals(asset_name)))
+    def get_all_balances(self) -> Dict[str, Decimal]:
+        return dict((asset_name, self.get_balance(asset_name))
                     for asset_name in self._raw_account_balances.keys())
 
     def get_raw_balance(self, asset_name: str) -> int:
         return self._raw_account_balances.get(asset_name, 0)
 
-    def get_balance(self, asset_name: str) -> float:
+    def get_balance(self, asset_name: str) -> Decimal:
         if asset_name not in self._raw_account_balances:
-            return 0.0
+            return s_decimal_0
         decimals: int = self.get_decimals(asset_name)
         raw_balance: int = self._raw_account_balances[asset_name]
-        return raw_balance * math.pow(10, -decimals)
+        raw_balance_in_decimal = Decimal(raw_balance)
+        balance_in_decimal = raw_balance_in_decimal / (10**decimals)
```

Comment:
Again, please use decimal arithmetics. Don't depend on the lucky fact that `10 ** integer` just happens to be an integer. It's pretty fragile.